### PR TITLE
Fix Build Hook Docs action

### PR DESCRIPTION
**Pull request by alexstine**
## Description

- Updated the actions/checkout version.
- Updated the actions/setup-node version.
- Switched actions/setup-node to use .nvmrc file for better version management.
- Gave the GITHUB_TOKEN proper write permissions to the repo.

## How Has This Been Tested?

Check for the passing action.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code is backward-compatible with WordPress 4.9 and PHP 7.0.
- [x] My code follows the WordPress coding standards.
- [x] My code has proper inline documentation.

### CODE REVIEW

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -8,12 +8,14 @@ on:
 jobs:
   hookdocs:
     runs-on: ubuntu-latest
+    permissions:
```
Comment (alexstine) [inline .github/workflows/build-docs.yml:11]: These permissions are required for write scope.

File: .github/workflows/build-docs.yml [hunk 2]
```diff
@@ -8,12 +8,14 @@ on:
...
+      contents: write
     steps:
-    - uses: actions/checkout@v1
```
Comment (alexstine) [inline .github/workflows/build-docs.yml:12]: Time for a newer version.

File: .github/workflows/build-docs.yml [hunk 3]
```diff
@@ -8,12 +8,14 @@ on:
...
-    - name: Use Node.js 10
```
Comment (alexstine) [inline .github/workflows/build-docs.yml:13]: This required a new version to work.

File: .github/workflows/build-docs.yml [hunk 4]
```diff
@@ -8,12 +8,14 @@ on:
...
-      uses: actions/setup-node@v1
+    - uses: actions/checkout@v3
+    - name: Use Node.js 14
+      uses: actions/setup-node@v3
       with:
-        node-version: '10.x'
+        node-version-file: '.nvmrc'
```
Comment (alexstine) [inline .github/workflows/build-docs.yml:18]: Use the file already providing our version.

**Review by alexstine — COMMENTED** [review]
Couple context notes.

**Review by joedolson — APPROVED** [review]

**Pull request merged**